### PR TITLE
techdebt: [DEL-1124]: close delegate down alert when they reconnect

### PR DESCRIPTION
**Pull request by arpit3107**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31210)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/scheduler/AlertCheckJob.java [hunk 1]
```diff
@@ -148,44 +138,28 @@ void checkForInvalidValidSMTP(String accountId) {
...
-    for (Delegate delegate : delegates) {
-      if (primaryConnections.contains(delegate.getUuid()) && isNotEmpty(delegate.getDelegateGroupName())) {
-        String delegateGroupName = delegate.getDelegateGroupName();
-        connectedScalingGroups.add(delegateGroupName);
-      }
+  private boolean isCGDelegate(Delegate delegate) {
+    try {
+      return !delegate.isNg();
+    } catch (NullPointerException e) {
+      // some old cg delegates don't have NG field, hence catching this NPE
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/scheduler/AlertCheckJob.java:161]: better to use delegate.isNg() !=null && delegate.isNg()

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31210)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/scheduler/AlertCheckJob.java, line 161 at r2](https://reviewable.io/reviews/harness/harness-core/31210#-MxYXW-JjOJXgCtdIfaJ-r2-161:-MxciWv49wgwSpgH4h3u:b-9tztzo) ([raw file](https://github.com/harness/harness-core/blob/5d1a0f5cb6890fe4c9edde93ee22fa1e17416592/400-rest/src/main/java/software/wings/scheduler/AlertCheckJob.java#L161)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

better to use delegate.isNg() !=null && delegate.isNg()
</blockquote></details>

!= operator cannot be applied to 'boolean' , 'null'

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31210)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31210)* status: :shipit: complete! ✓

**Pull request merged**